codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerDowngrader.java

```diff
@@ -78,9 +80,10 @@ protected void decode(ChannelHandlerContext ctx, Http2StreamFrame frame, List<Ob
                     out.add(full);
                 }
             } else {
-                out.add(HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders));
+                HttpRequest req = HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders);
+                HttpUtil.setTransferEncodingChunked(req, true);
```

Comment:
content-less requests should be OK, since the header will be sent with an "end of stream" marker.  content-length-y headers are definitely in danger though, I'll inspect for content-length first.